src/runtime/eval.ml

```diff
@@ -370,6 +382,7 @@ and check_judgement ({Location.it=c'; at} as c) bdry =
   | Syntax.Lookup _
   | Syntax.Update _
   | Syntax.Fresh _
+  | Syntax.AbstractAtom _
```

Comment:
Yes, this could be improved, it would be mostly like `Syntax.Sequence` except that `c1` will be required to be an atom.

Reply:
Let's move this to a new issue, as it is an improvement (rather than a fix).